### PR TITLE
Wip/mk/scatter plot visual improvements

### PR DESCRIPTION
**Pull request by marthasharkey**
This adds multi series support to the scatter plot 

- all numeric columns will be plotted with either column 'X' or the left most numeric column being used as the x-axis 
- colours are set as the default d3 colours 
- a legend at the top of the plot shows what colour is which 'series' or column 
- the legends labels will be cropped at 10 chars but has a tooltip for full column name 

The plot now has, by default, a 'normal' cursor to allow the user to hover over points to show a tooltip whoch displays it's details.
Clicking the '+' button changes the cursor to use the d3 brush allowing the user to zoom in on parts of the plot

10933-scatter-plot-v1

image

10933-scatter-plot-v1-1

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

File: app/gui2/src/components/visualizations/ScatterplotVisualization.vue [hunk 1]
```diff
@@ -155,7 +157,10 @@ const data = computed<Data>(() => {
   }
   const points = rawData.points ?? { labels: 'visible' }
   const focus: Focus | undefined = rawData.focus
-  return { axis, points, data, focus }
+  // eslint-disable-next-line camelcase
+  const is_multi_series: boolean = Boolean(rawData.is_multi_series)
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/ScatterplotVisualization.vue]: `!!rawData.is_multi_series`

File: app/gui2/src/components/visualizations/ScatterplotVisualization.vue [hunk 2]
```diff
@@ -178,6 +185,10 @@ const focus = ref<Focus>()
 const shouldAnimate = ref(false)
 const xDomain = ref([0, 1])
 const yDomain = ref([0, 1])
+const selectionEnabled = ref(false)
+const enableSelection = () => {
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/ScatterplotVisualization.vue]: - The name says `enable`, but the implementation toggles.
- I don't see a need to wrap this in a function. It's a one liner, and is unlikely to need future changes--if any logic needs to be run when the value is changed, the Vue solution is to use watchers.

File: app/gui2/src/components/visualizations/ScatterplotVisualization.vue [hunk 3]
```diff
@@ -380,18 +406,38 @@ function startZoom(event: d3.D3ZoomEvent<Element, unknown>) {
   actionStartYScale = yScale.value.copy()
 }
 
-const brush = computed(() =>
-  d3
+const brush = computed(() => {
+  selectionEnabled.value
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/ScatterplotVisualization.vue]: This is not doing anything

File: app/gui2/src/components/visualizations/ScatterplotVisualization.vue [hunk 4]
```diff
@@ -539,16 +672,18 @@ useEvent(document, 'keydown', bindings.handler({ zoomToSelected: () => zoomToSel
 <template>
   <VisualizationContainer :belowToolbar="true">
     <template #toolbar>
+      <SvgButton name="add" title="Enable Selection" @click="enableSelection()" />
```
Comment (kazcw) [inline app/gui2/src/components/visualizations/ScatterplotVisualization.vue]: I think it's confusing to use the same icon as the add-node buttons for something unrelated.

**Review by kazcw — CHANGES_REQUESTED** [review]

File: app/gui2/src/components/visualizations/ScatterplotVisualization.vue (hunk 4 above)
Comment (marthasharkey) [reply app/gui2/src/components/visualizations/ScatterplotVisualization.vue]: Yes I agree! I'll talk to Ad about what icon is best!

**Review by kazcw — APPROVED** [review]

File: distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso [hunk 5]
```diff
@@ -216,13 +217,14 @@ json_from_table table bounds limit =
     data = table.point_data fields_for_plot |> bound_data bounds |> limit_data limit
     fields_for_axes = get_axes_field number_of_numeric_cols
     axes = table.axes fields_for_axes
-    JS_Object.from_pairs [[data_field, data], [axis_field, axes]] . to_json
+    is_mulit_series = number_of_numeric_cols > 0
+    JS_Object.from_pairs [[data_field, data], [axis_field, axes], ["is_multi_series", is_mulit_series]] . to_json
```
Comment (jdunkerley) [inline distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso]:
```suggestion
    is_multi_series = number_of_numeric_cols > 0
    JS_Object.from_pairs [[data_field, data], [axis_field, axes], ["is_multi_series", is_multi_series]] . to_json
```

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**